src/main/java/WayofTime/bloodmagic/compat/jei/alchemyArray/AlchemyArrayCraftingRecipeJEI.java

```diff
@@ -13,22 +11,31 @@
 public class AlchemyArrayCraftingRecipeJEI extends BlankRecipeWrapper
 {
     @Nonnull
-    private final List<ItemStack> inputs;
+    private final Object inputs;
+
+    @Nonnull
+    private final ItemStack catalyst;
 
     @Nonnull
     private final ItemStack output;
 
     @SuppressWarnings("unchecked")
-    public AlchemyArrayCraftingRecipeJEI(@Nonnull ItemStack input, @Nullable ItemStack catalyst, @Nonnull ItemStack output)
+    public AlchemyArrayCraftingRecipeJEI(@Nonnull Object input, @Nonnull ItemStack catalyst, @Nonnull ItemStack output)
     {
```

Comment:
Trust me, there's a reason it looks that ugly. I wouldn't have done it, otherwise. If you want it to be counted as an actual input in JEI, you *must* pass it through `getInputs()`.